core/src/main/scala/spinal/core/FixedPoint.scala

```diff
@@ -883,6 +886,236 @@ class Fix(val intWidth: Int, val bitWidth: Int, val signed: Boolean) extends Mul
   override def toString: String = s"${component.getPath() + "/" + this.getDisplayName()} : ${getClass.getSimpleName}[signed=${if (this.signed) "1" else "0"},whole=${this.wholeWidth},frac=${this.fracWidth}]"
 }
 
+object AutoFix {
+
+  def apply(f: Fix): AutoFix = {
+    val maxValue = BigDecimal(2).pow(f.wholeWidth)-BigDecimal(2).pow(-f.fracWidth)
+    val minvalue = if (f.signed) -BigDecimal(2).pow(f.wholeWidth) else BigDecimal(0)
+    val ret = new AutoFix(maxValue, minvalue, -f.fracWidth exp)
+    ret.raw := f.raw
+    ret
+  }
+
+}
+
+class AutoFix(val maxValue: BigDecimal, val minValue: BigDecimal, val exp: ExpNumber) extends MultiData {
```

Comment:
Hi ^^

I would say, use BigInt instead of BigDecimal, as BigInt is just the raw storage of a number, while BigDecimal as some "funky" stuff attached.